metaflow/plugins/kfp/kfp.py

```diff
@@ -356,6 +363,15 @@ def build_kfp_component(node: DAGNode, task_id: str) -> KfpComponent:
                     ),
                     None,  # default
                 ),
+                environment_decorator=next(
+                    (
+                        deco
+                        for deco in node.decorators
+                        if isinstance(deco, EnvironmentDecorator)
+                        and "kubernetes_vars" in deco.attributes
```

Comment:
Maybe I'm missing something obvious here but how does the `kubernetes_vars` attribute actually get set? Is there some magic that is happening that the diffs aren't exposing, from what I can tell I cannot see how that attribute is written 😅

Reply:
True, it is magic somewhere deep in the Metaflow decorators code.
https://github.com/zillow/metaflow/blob/84537811c14f406925d7a2e41363565693fb1971/metaflow/decorators.py#L87-L93

Reply:
Ah this makes more sense now I see [this thread](https://github.com/zillow/metaflow/pull/74/files#r617256097) from @cloudw, looks like as he described this isn't automatically populated yet but likely could/should be?

Reply:
I responded to the thread, they get added as part of this PR in https://github.com/zillow/metaflow/blob/cc1e5318f610c77b4cee237dc037510c35d292cc/metaflow/plugins/kfp/kfp.py#L715-L721

`kubernetes_vars` are compile time variables rendered by k8s at runtime into the pod by k8s.

The normal env variables are populated by step_init at compile time and injected in this step (which in our case we let k8s do for us): https://github.com/zillow/metaflow/blob/cc1e5318f610c77b4cee237dc037510c35d292cc/metaflow/plugins/environment_decorator.py#L44-L45